src/ll/request.rs

```diff
@@ -43,378 +136,1216 @@ impl fmt::Display for RequestError {
-    Init {
-        arg: &'a fuse_init_in,
-    },
-    OpenDir {
-        arg: &'a fuse_open_in,
-    },
-    ReadDir {
-        arg: &'a fuse_read_in,
-    },
-    ReleaseDir {
-        arg: &'a fuse_release_in,
-    },
-    FSyncDir {
-        arg: &'a fuse_fsync_in,
-    },
-    GetLk {
-        arg: &'a fuse_lk_in,
-    },
-    SetLk {
-        arg: &'a fuse_lk_in,
-    },
-    SetLkW {
-        arg: &'a fuse_lk_in,
-    },
-    Access {
-        arg: &'a fuse_access_in,
-    },
-    Create {
+    }
+    impl<'a> RemoveXAttr<'a> {
+        pub fn name(&self) -> &'a OsStr {
+            self.name
+        }
+    }
+    #[derive(Debug)]
+    pub struct Flush<'a>(&'a fuse_flush_in);
+    impl<'a> Flush<'a> {
+        pub fn file_handle(&self) -> FileHandle {
+            FileHandle(self.0.fh)
+        }
+        pub fn lock_owner(&self) -> LockOwner {
+            LockOwner(self.0.lock_owner)
+        }
+    }
+    #[derive(Debug)]
+    pub struct Init<'a>(&'a fuse_init_in);
+    impl<'a> Init<'a> {
+        pub fn capabilities(&self) -> u32 {
+            self.0.flags
+        }
+        pub fn max_readahead(&self) -> u32 {
+            self.0.max_readahead
+        }
+        pub fn version(&self) -> (u32, u32) {
+            (self.0.major, self.0.minor)
+        }
+    }
+    #[derive(Debug)]
+    pub struct OpenDir<'a>(&'a fuse_open_in);
+    impl<'a> OpenDir<'a> {
+        /// Flags as passed to open
+        pub fn flags(&self) -> i32 {
+            self.0.flags
+        }
+    }
+    #[derive(Debug)]
+    pub struct ReadDir<'a>(&'a fuse_read_in);
+    impl<'a> ReadDir<'a> {
+        pub fn file_handle(&self) -> FileHandle {
+            FileHandle(self.0.fh)
+        }
+        pub fn offset(&self) -> i64 {
+            self.0.offset
+        }
+        pub fn size(&self) -> u32 {
+            self.0.size
+        }
+    }
+    #[derive(Debug)]
+    pub struct ReleaseDir<'a>(&'a fuse_release_in);
+    impl<'a> ReleaseDir<'a> {
+        pub fn file_handle(&self) -> FileHandle {
+            FileHandle(self.0.fh)
+        }
+        pub fn flush(&self) -> bool {
+            self.0.release_flags & consts::FUSE_RELEASE_FLUSH != 0
+        }
+        pub fn lock_owner(&self) -> Option<LockOwner> {
+            #[cfg(not(feature = "abi-7-17"))]
+            return Some(LockOwner(self.0.lock_owner));
+            #[cfg(feature = "abi-7-17")]
+            if self.0.release_flags & FUSE_RELEASE_FLOCK_UNLOCK != 0 {
+                Some(LockOwner(self.0.lock_owner))
+            } else {
+                None
+            }
+        }
+        /// TODO: Document what values this may take
+        pub fn flags(&self) -> i32 {
+            self.0.flags
+        }
+    }
+    #[derive(Debug)]
+    pub struct FSyncDir<'a>(&'a fuse_fsync_in);
+    impl<'a> FSyncDir<'a> {
+        pub fn file_handle(&self) -> FileHandle {
+            FileHandle(self.0.fh)
+        }
+        pub fn datasync(&self) -> bool {
+            self.0.fsync_flags & 1 != 0
```

Comment:
This should use the new `FUSE_FSYNC_FDATASYNC` constant you added, instead of `1`

Reply:
Fixed, good spot.  I could have sworn that I'd already done that :)